core/src/main/java/com/sequenceiq/cloudbreak/converter/StackToTemplatePreparationObjectConverter.java

```diff
@@ -381,10 +381,11 @@ private void decorateBuilderWithAccountMapping(StackDtoDelegate source, Detailed
 
     private void decorateDatalakeView(StackView source, TemplatePreparationObject.Builder builder) {
         DatalakeView datalakeView = null;
-        if (StringUtils.isNotEmpty(source.getEnvironmentCrn()) && StackType.WORKLOAD.equals(source.getType())) {
```

Comment:
you should keep this condition, and add stack CRN to templatePreparationObject from stack instead, then you can use that in case of RangerRazDatalakeProvider.
in case of datahub, it is fine to add DL CRN to datalakeView